samples/csharp_dotnetcore/23.facebook-events/Dialogs/MainDialog.cs

```diff
@@ -0,0 +1,130 @@
+        {
+            Logger.LogInformation("MainDialog.ChoiceStepAsync");
+            var turnContext = stepContext.Context;
+
+            var text = turnContext.Activity.Text;
+            // If the text is one of the quick reply options, continue to the ShowResultStep
+            if (new[] { QuickRepliesOption, FacebookPageNameOption, PostBackOption }.Contains(text))
+            {
+                return await stepContext.NextAsync();
+            }
+            else
+            {
+                return await ShowChoices(stepContext, cancellationToken);
+            }
+        }
+
+        private static async Task<DialogTurnResult> ShowChoices(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            // Create options for the prompt
+            var options = new PromptOptions()
+            {
+                Prompt = stepContext.Context.Activity.CreateReply("What Facebook feature would you like to try? Here are some quick replies to choose from!"),
+                Choices = new List<Choice>(),
+            };
+
+            // Add the choices for the prompt.
+            options.Choices.Add(new Choice() { Value = QuickRepliesOption, Action = new CardAction() { Title = QuickRepliesOption, Type = ActionTypes.PostBack, Value = QuickRepliesOption } });
+            options.Choices.Add(new Choice() { Value = FacebookPageNameOption, Action = new CardAction() { Title = FacebookPageNameOption, Type = ActionTypes.PostBack, Value = FacebookPageNameOption } });
+            options.Choices.Add(new Choice() { Value = PostBackOption, Action = new CardAction() { Title = PostBackOption, Type = ActionTypes.PostBack, Value = PostBackOption } });
+
+            return await stepContext.PromptAsync(nameof(ChoicePrompt), options, cancellationToken);
+        }
+
+        private async Task<DialogTurnResult> ShowResultStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        {
+            Logger.LogInformation("MainDialog.ShowResultStep");
+            var turnContext = stepContext.Context;
+
+            // Initially the bot offers to showcase 3 Facebook features: Quick replies, PostBack and getting the Facebook Page Name.
+            switch (turnContext.Activity.Text)
+            {
+                // Here we showcase how to obtain the Facebook page name.
+                // This can be useful for the Facebook multi-page support provided by the Bot Framework.
+                // The Facebook page name from which the message comes from is in turnContext.Activity.Recipient.Name.
+                case FacebookPageNameOption:
+                    {
+                        var reply = turnContext.Activity.CreateReply($"This message comes from the following Facebook Page: {turnContext.Activity.Recipient.Name}");
+                        await turnContext.SendActivityAsync(reply);
```

Comment:
Pass cancellationToken to SendActivityAsync()

Reply:
addressed